src/game/client.cpp

```diff
@@ -291,6 +291,99 @@ namespace client
     }
     ICOMMAND(0, writevars, "sii", (char *name, int *all, int *sv), if(!(identflags&IDF_WORLD)) writegamevars(name, *all!=0, *sv!=0));
 
+    void write_weapon_vars(const char* weapon_name, const char* file_name, bool all = false, bool server = false)
+    {
+        if (!weapon_name || !*weapon_name)
+        {
+            conoutft(CON_EVENT, "\frNo weapon name specified");
+            return;
+        }
+        
+        if (!file_name || !*file_name)
+        {
+            conoutft(CON_EVENT, "\fmNo filename specified, using 'vars.cfg'");
+            file_name = "vars.cfg";
+        }
+        
+        stream* file_stream = openfile(file_name, "w");
+        if (!file_stream) 
+        {
+            conoutft(CON_EVENT, "\frFailed to open file");
+            return;
+        }
+        
+        vector<ident*> identifiers;
+        enumerate(idents, ident, identifier, identifiers.add(&identifier));
+        identifiers.sortname();
+        
+        for (int i = 0; i < identifiers.length(); i++)
+        {
+            ident& identifier = *identifiers[i];
+            
+            // skip if identifier is not client sided or readonly,
+            // a worldvar or doesn't start with the weapon_name
+            if ( !(identifier.flags & IDF_CLIENT)
+               || (identifier.flags & IDF_READONLY)
+               || (identifier.flags & IDF_WORLD)
+               || (strncmp(identifier.name, weapon_name, strlen(identifier.name)) == 0))
```

Comment:
This line is not correct. It means that the identifier will be skipped when the identifier name is equal to the weapon name. You want the identifier to be skipped when the identifier name does not begin with the weapon name.

Reply:
well yes, but actually no. for some reason it saves all changed variables

Reply:
That is because of this last test. It does not actually filter by weapon name.

Reply:
yeah I figured that out, do you have any idea how this could be achieved?

Reply:
One way (keeping with C string methods) is to replace
`|| (strncmp(identifier.name, weapon_name, strlen(identifier.name)) == 0)`
with 
`|| (strncmp(identifier.name, weapon_name, strlen(weapon_name)) != 0)`

Which means: continue if the identifier name does not begin with the weapon name.

Reply:
ye, I also messed up the strlen, it has to be strlen(weapon_name), not strlen(identifier.name). heh.